spec/unit/facter/login_defs_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+
+describe "custom fact login_defs" do
+
+  before(:each) do
+    Facter.clear
+
+    Facter.stubs(:value).with(:operatingsystem).returns('Linux')
+  end
+
+  context 'with a well formed /etc/login.defs' do
+    let(:login_defs_content) { <<-EOM
+MAIL_DIR        /var/spool/mail
+PASS_MAX_DAYS   99999
+PASS_MIN_DAYS   0
+PASS_MIN_LEN    5
+PASS_WARN_AGE   7
+UID_MIN         1000
+UID_MAX         60000
+SYS_UID_MIN     201
+SYS_UID_MAX     999
+GID_MIN         1000
+GID_MAX         60000
+SYS_GID_MIN     201
+SYS_GID_MAX     999
+CREATE_HOME     yes
+UMASK           077
+USERGROUPS_ENAB yes
+ENCRYPT_METHOD  SHA512
+MD5_CRYPT_ENAB  no
+      EOM
+    }
+
+    it 'should return hash of values from /etc/login.defs with appropriate conversions' do
+      File.expects(:exist?).with('/etc/login.defs').returns(true)
+      File.expects(:readable?).with('/etc/login.defs').returns(true)
+      File.expects(:read).with('/etc/login.defs').returns(login_defs_content)
+
+      File.stubs(:exist?).with(Not(equals('/etc/login.defs')))
```

Comment:
Given how hard this was to figure out how to mock the File.* commands with mocha when these commands are called in parts of the code that is not under test with different arguments, seems like a comment would be warranted.